fastserde/avro-fastserde-tests-common/src/test/java/com/linkedin/avro/fastserde/FastSerdeCacheTest.java

```diff
@@ -68,4 +80,81 @@ public void testBuildFastSpecificDeserializerWithCorrectClasspath() {
     FastSerdeCache cache = FastSerdeCache.getDefaultInstance();
     cache.buildFastSpecificDeserializer(TestRecord.SCHEMA$, TestRecord.SCHEMA$);
   }
+
+  @Test(groups = "serializationTest", timeOut = 5_000L,
+          expectedExceptions = UnsupportedOperationException.class,
+          expectedExceptionsMessageRegExp = "Fast specific serializer could not be generated.")
+  public void testSpecificSerializationFailsFast() throws Exception {
+    serializationShouldFailFast(FastSpecificDatumWriter::new);
+  }
+
+  @Test(groups = "serializationTest", timeOut = 5_000L,
+          expectedExceptions = UnsupportedOperationException.class,
+          expectedExceptionsMessageRegExp = "Fast generic serializer could not be generated.")
+  public void testGenericSerializationFailsFast() throws Exception {
+    serializationShouldFailFast(FastGenericDatumWriter::new);
+  }
+
+  @Test(groups = "deserializationTest", timeOut = 5_000L,
+          expectedExceptions = UnsupportedOperationException.class,
+          expectedExceptionsMessageRegExp = "Fast specific deserializer could not be generated.")
+  public void testSpecificDeserializationFailsFast() throws Exception {
+    deserializationShouldFailFast(FastSpecificDatumReader::new);
+  }
+
+  @Test(groups = "deserializationTest", timeOut = 5_000L,
+          expectedExceptions = UnsupportedOperationException.class,
+          expectedExceptionsMessageRegExp = "Fast generic deserializer could not be generated.")
+  public void testGenericDeserializationFailsFast() throws Exception {
+    deserializationShouldFailFast(FastGenericDatumReader::new);
+  }
```

Comment:
You can use data provider to avoid these test classes.
https://testng.org/doc/documentation-main.html#parameters-dataproviders

Reply:
I was thinking about it but at the end I think current way is easier to read.
Please keep in mind we would need data providers (it's 2x2 not 4x1). These two methods have too much differences to unify them (without loosing readability):
```
serializationShouldFailFast
deserializationShouldFailFast
```

There other thing is data-providers would force explicit try-catch blocks instead of annotation's attributes:
```
expectedExceptions = UnsupportedOperationException.class,
expectedExceptionsMessageRegExp = "Fast specific deserializer could not be generated."
```